src/pages/ReportDetailsPage.tsx

```diff
@@ -1,103 +1,103 @@
+    const isPolicyMember = useMemo(() => PolicyUtils.isPolicyMember(report?.policyID ?? '', policies), [report?.policyID, policies]);
+    const shouldUseFullTitle = useMemo(() => ReportUtils.shouldUseFullTitleToDisplay(report), [report]);
+    const isChatRoom = useMemo(() => ReportUtils.isChatRoom(report), [report]);
+    const isThread = useMemo(() => ReportUtils.isChatThread(report), [report]);
+    const isUserCreatedPolicyRoom = useMemo(() => ReportUtils.isUserCreatedPolicyRoom(report), [report]);
+    const isArchivedRoom = useMemo(() => ReportUtils.isArchivedRoom(report), [report]);
+    const isMoneyRequestReport = useMemo(() => ReportUtils.isMoneyRequestReport(report), [report]);
 
     // eslint-disable-next-line react-hooks/exhaustive-deps -- policy is a dependency because `getChatRoomSubtitle` calls `getPolicyName` which in turn retrieves the value from the `policy` value stored in Onyx
-    const chatRoomSubtitle = useMemo(() => ReportUtils.getChatRoomSubtitle(props.report), [props.report, policy]);
-    const parentNavigationSubtitleData = ReportUtils.getParentNavigationSubtitle(props.report);
-    const participants = useMemo(() => ReportUtils.getVisibleMemberIDs(props.report), [props.report]);
+    const chatRoomSubtitle = useMemo(() => ReportUtils.getChatRoomSubtitle(report), [report, policy]);
+    const parentNavigationSubtitleData = ReportUtils.getParentNavigationSubtitle(report);
+    const participants = useMemo(() => ReportUtils.getVisibleMemberIDs(report), [report]);
 
-    const isGroupDMChat = useMemo(() => ReportUtils.isDM(props.report) && participants.length > 1, [props.report, participants.length]);
+    const isGroupDMChat = useMemo(() => ReportUtils.isDM(report) && participants.length > 1, [report, participants.length]);
 
-    const isPrivateNotesFetchTriggered = !_.isUndefined(props.report.isLoadingPrivateNotes);
+    const isPrivateNotesFetchTriggered = report?.isLoadingPrivateNotes !== undefined;
 
     useEffect(() => {
         // Do not fetch private notes if isLoadingPrivateNotes is already defined, or if network is offline.
-        if (isPrivateNotesFetchTriggered || props.network.isOffline) {
+        if (isPrivateNotesFetchTriggered || isOffline) {
             return;
         }
 
-        Report.getReportPrivateNote(props.report.reportID);
-    }, [props.report.reportID, props.network.isOffline, isPrivateNotesFetchTriggered]);
+        Report.getReportPrivateNote(report?.reportID ?? '');
+    }, [report?.reportID, isOffline, isPrivateNotesFetchTriggered]);
 
-    const menuItems = useMemo(() => {
+    const menuItems: ReportDetailsPageMenuItem[] = useMemo(() => {
         const items = [];
 
         if (!isGroupDMChat) {
             items.push({
                 key: CONST.REPORT_DETAILS_MENU_ITEM.SHARE_CODE,
-                translationKey: 'common.shareCode',
+                translationKey: 'common.shareCode' as const,
```

Comment:
If you add type to items, you don't need `as const`
```suggestion
                translationKey: 'common.shareCode',
```